prov/terraform/google/generator.go

```diff
@@ -117,34 +117,44 @@ func (g *googleGenerator) GenerateTerraformInfraForNode(ctx context.Context, cfg
 		return false, nil, nil, err
 	}
 	outputs := make(map[string]string)
-	var instances []string
-	switch nodeType {
-	case "yorc.nodes.google.Compute":
-		instances, err = deployments.GetNodeInstancesIds(kv, deploymentID, nodeName)
+	instances, err := deployments.GetNodeInstancesIds(kv, deploymentID, nodeName)
+	if err != nil {
+		return false, nil, nil, err
+	}
+
+	for instNb, instanceName := range instances {
+		instanceState, err := deployments.GetInstanceState(kv, deploymentID, nodeName, instanceName)
 		if err != nil {
 			return false, nil, nil, err
 		}
+		if instanceState == tosca.NodeStateDeleting || instanceState == tosca.NodeStateDeleted {
+			// Do not generate something for this node instance (will be deleted if exists)
+			continue
+		}
 
-		for index, instanceName := range instances {
-			var instanceState tosca.NodeState
-			instanceState, err = deployments.GetInstanceState(kv, deploymentID, nodeName, instanceName)
+		switch nodeType {
+		case "yorc.nodes.google.Compute":
+			instances, err = deployments.GetNodeInstancesIds(kv, deploymentID, nodeName)
 			if err != nil {
 				return false, nil, nil, err
 			}
-			if instanceState == tosca.NodeStateDeleting || instanceState == tosca.NodeStateDeleted {
-				// Do not generate something for this node instance (will be deleted if exists)
-				continue
+
+			err = g.generateComputeInstance(ctx, kv, cfg, deploymentID, nodeName, instanceName, instNb, &infrastructure, outputs)
+			if err != nil {
+				return false, nil, nil, err
 			}
-			err = g.generateComputeInstance(ctx, kv, cfg, deploymentID, nodeName, instanceName, index, &infrastructure, outputs)
+
+		case "yorc.nodes.google.PublicNetwork":
```

Comment:
Should we take care of this at all in yorc engine?
Those types should be defined only in the yorc-a4c-plugin and transformed into a `Address` by a modifier.

Reply:
That's correct